### ISSUE TITLE
[FormsyDate] formatDate prop skips locale 

### ISSUE THREAD

**Issue by lukkuz** [mbrookes/formsy-material-ui#31]
Current implementation of formatDate formatter skips locale settings thus giving incorrect value due time-zone calculations.

Date set for today is 2015-11-30 which gives "2015-11-29T23:00:00.000Z" as ISO representation. My timezone is CET. Substringing it will give the day before which is incorrect.

The simplest fix would be to change FormsyDate.jsx:12 from

```javascript
formatDate={(date) => date.toISOString().substring(0,10)}
```
to
```javascript
formatDate={(date) => date.toLocaleString().substring(0,10)}
```

**Comment by mbrookes** [comment]
Good catch - my attempt to make DateField more international-friendly by default by using ISO8601 rather than US date format has clearly backfired! :smile:

In the meantime, as a workaround you can of course override `formatDate`, just  as you would have had to anyway.

As for a fix, `toLocaleDateString()` seems like it ought to help, but it defaults to a US date format unless provided with a locale, so unless there's a local that produces ISO8601 style YYYY-MM-DD format, it doesn't quite solve the problem.

There's `Intl.DateTimeFormat`, but it isn't supported by all browsers, and pulling in a polyfill just for a default designed to be overridden seems overkill.

The best I've come up with so far is was this slightly hacky workaround:

```
formatDate={(date) => (new Date(date.toDateString()+" 12:00:00 +0000")).toISOString().substring(0,10)}
```

Could you check in the browser if that seems to solves the problem for you?

The alternative would be to just use the output of toDateString directly, which, like ISO8601, is unambiguous: "Tue Dec 01 2015", but has the disadvantage that it's in English.

**Comment by lukkuz** [comment]
Yes, it works perfectly fine. Below is the transcript from the test inside my browser.
```javascript
> var date = new Date(2015, 11, 1)
<- undefined
> date.toString()
<- "Tue Dec 01 2015 00:00:00 GMT+0100 (Środkowoeuropejski czas stand.)"
> date.toISOString()
<- "2015-11-30T23:00:00.000Z"
> (new Date(date.toDateString()+" 12:00:00 +0000")).toISOString().substring(0,10)
<- "2015-12-01"
```

**Comment by mbrookes** [comment]
Could you check against this branch? https://github.com/mbrookes/formsy-material-ui/tree/default-date

Depending on how you import NPMs, you should be able to reference the brach or the specific commit: 43f4955962774edff066abb4ca864b27dc14fdf2

**Comment by lukkuz** [comment]
I've tried checking against this commit but things seem to be broken. My build fails with a message:
>ERROR in ./src/components/Comment/CommentDialog.js
>Module not found: Error: Cannot resolve module 'formsy-material-ui' in /home/osboxes/devel/tmp-web-client/src/components/Comment
>...skipped some lines...
>resolve file
>              /home/osboxes/devel/tmp-web-client/node_modules/formsy-material-ui/index.js doesn't exist

I'm using ES6 syntax in my code.

My understanding is that somehow package.json file was also edited and ``` "main": "lib/index.js"``` was replaced with ``` "main": "index.js"``` and there is no such file in main dir.

I've tried to find a commit that did this thing but wasn't successful. It seems that version 0.3.2 was released from version than had not been pushed into main repo. I say that because I can't see ```"gitHead": "e43fd8fec546c68766495e154bf3d81393f38dd7"``` (package.json:53 from formsy-material-ui@0.3.2) in repo which is strange. No such commit.

**Comment by mbrookes** [comment]
You're right - it looks like I published from the `separate-components` branch instead of merging and publishing from master. Rather than try to reconcile now, I've updated this `default-date` branch with what should be a working package.json (and a transpiled version of FormsyDate).

If it looks okay for you, then I'll try to clean up my branch / commit foobar and merge this with master and publish.
